### ISSUE TITLE
Rollback seems to be requested for test jobs?

### ISSUE THREAD

**Issue by vinjana** [eilslabs/Roddy#133]
I get the following output, if a rollback is initialized:

```
$ /basePath/RoddyProject_Roddy2.4/Roddy/roddy.sh rerun testCoBaseConfigs-Roddy-2-4.Picard.SoftwareBwa.WGS@alignment testpid --useconfig=/basePath//RoddyProject_Roddy2.4/testConfigs/applicationProperties-analysis-local-lsf.ini --usefeaturetoggleconfig=/basePath//RoddyProject_Roddy2.4/testConfigs/featureToggles.ini --configurationDirectories=/basePath//RoddyProject_Roddy2.4/configs,/basePath//RoddyProject_Roddy2.4/testConfigs/AlignmentAndQCWorkflows/,/basePath//RoddyProject_Roddy2.4/testConfigs/AlignmentAndQCWorkflows//resources --usePluginVersion=AlignmentAndQCWorkflows:current --useiodir=/basePath/testData-small//view-by-pid,/basePath/tests/AlignmentAndQCWorkflows-OTPConfig-1.5-Roddy-2.4/testCoBaseConfigs-Roddy-2-4.Picard.SoftwareBwa.WGS --cvalues=INDEX_PREFIX:/icgc/dkfzlsdf/dmg/otp/production/processing/reference_genomes/bwa06_1KGRef/hs37d5.fa,CHROM_SIZES_FILE:/icgc/dkfzlsdf/dmg/otp/production/processing/reference_genomes/bwa06_1KGRef/stats/hs37d5.fa.chrLenOnlyACGT_realChromosomes.tab,usedResourcesSize:xs,runFingerprinting:true,outputAllowAccessRightsModification:false,runACEseqQc:true,cnv_min_coverage_ALN:1,mapping_quality_ALN:0,min_windows_ALN:1,runFastQC:true,RODDY_SCRATCH:/local/${USER}/${RODDY_JOBID},workflowEnvironmentScript:workflowEnvironment_tbiLsf --useRoddyVersion=current --additionalImports=wgs-standard-lsf-fpga-0.7.8

The plugin AlignmentAndQCWorkflows [ Version: current ] was loaded (/basePath/RoddyProject_Roddy2.4/plugins_R2.4/AlignmentAndQCWorkflows).
The plugin COWorkflowsBasePlugin [ Version: current ] was loaded (/basePath/RoddyProject_Roddy2.4/plugins_R2.4/COWorkflowsBasePlugin).
The plugin PluginBase [ Version: current ] was loaded (/basePath/RoddyProject_Roddy2.4/plugins_R2.4/PluginBase).
The plugin DefaultPlugin [ Version: current ] was loaded (/basePath/RoddyProject_Roddy2.4/plugins_R2.4/DefaultPlugin).
  Fully load configurationFile /basePath/RoddyProject_Roddy2.4/testConfigs/AlignmentAndQCWorkflows/testCoBaseConfigs-roddy-2.4.xml
  Fully load configurationFile /basePath/RoddyProject_Roddy2.4/testConfigs/AlignmentAndQCWorkflows/testCoBaseConfigs-roddy-2.4.xml
  Fully load configurationFile /basePath/RoddyProject_Roddy2.4/testConfigs/AlignmentAndQCWorkflows/testCoBaseConfigs-roddy-2.4.xml
  Fully load configurationFile /basePath/RoddyProject_Roddy2.4/testConfigs/AlignmentAndQCWorkflows/testCoBaseConfigs-roddy-2.4.xml
  Fully load configurationFile /basePath/RoddyProject_Roddy2.4/plugins_R2.4/DefaultPlugin/resources/configurationFiles/default.xml
  Fully load configurationFile /basePath/RoddyProject_Roddy2.4/configs/coBaseProject.xml
  Fully load configurationFile /basePath/RoddyProject_Roddy2.4/plugins_R2.4/DefaultPlugin/resources/configurationFiles/default.xml
  Fully load configurationFile /basePath/RoddyProject_Roddy2.4/configs/cofilenames.xml
  Fully load configurationFile /basePath/RoddyProject_Roddy2.4/configs/coApplicationsAndReferenceFiles.xml
  Fully load configurationFile /basePath/RoddyProject_Roddy2.4/plugins_R2.4/AlignmentAndQCWorkflows/resources/configurationFiles/analysisQC.xml
  Fully load configurationFile /basePath/RoddyProject_Roddy2.4/plugins_R2.4/COWorkflowsBasePlugin/resources/configurationFiles/commonCOWorkflowsSettings.xml
  Fully load configurationFile /basePath/RoddyProject_Roddy2.4/testConfigs/AlignmentAndQCWorkflows/resources/wgs-standard-lsf-fpga-0.7.8.xml
Found 3 datasets in the in- and output directories.
Found 1 samples for dataset testpid
Searching for lane files in directory /basePath/testData-small/view-by-pid/testpid/tumor/paired
Processed sample tumor and found 2 groups of lane files.
Tried to call a non generic tool via the generic call method
A workflow error occurred, try to rollback / abort submitted jobs.
bkill 0x88FBF8B0F98A8 0x88FBF8C21B1B5 0x88FBF91065209 0x88FBF91A4F17F 0x88FBF9290F166 0x88FBF9464C3D2 0x88FBFAEE44660 0x88FBFB0F56E55 0x88FBFB269C380 0x88FBFB4D6C618
An unknown / unhandled exception occurred: 'Not able to context tool coveragePlotSingle'
de.dkfz.roddy.knowledge.methods.GenericMethod._callGenericToolOrToolArray(GenericMethod.groovy:232)
de.dkfz.roddy.knowledge.methods.GenericMethod.callGenericTool(GenericMethod.groovy:50)
de.dkfz.b080.co.files.CoverageTextFile.plot(CoverageTextFile.java:49)
de.dkfz.b080.co.files.CoverageTextFileGroup.plot(CoverageTextFileGroup.java:47)
de.dkfz.b080.co.qcworkflow.QCPipeline.execute(QCPipeline.groovy:94)
de.dkfz.roddy.core.ExecutionContext.execute(ExecutionContext.groovy:625)
de.dkfz.roddy.core.Analysis.executeRun(Analysis.java:397)
de.dkfz.roddy.core.Analysis.run(Analysis.java:206)
de.dkfz.roddy.client.cliclient.RoddyCLIClient.rerun(RoddyCLIClient.groovy:493)
de.dkfz.roddy.client.cliclient.RoddyCLIClient.parseStartupMode(RoddyCLIClient.groovy:114)
de.dkfz.roddy.Roddy.parseRoddyStartupModeAndRun(Roddy.java:688)
de.dkfz.roddy.Roddy.startup(Roddy.java:282)
de.dkfz.roddy.Roddy.main(Roddy.java:214)
Found 1 samples for dataset testpid
Processed sample tumor and found 2 groups of lane files.
Creating the following execution directory to store information about this process:
        /basePath/tests/AlignmentAndQCWorkflows-OTPConfig-1.5-Roddy-2.4/testCoBaseConfigs-Roddy-2-4.Picard.SoftwareBwa.WGS/testpid/roddyExecutionStore/exec_171019_121111722_kensche_alignment
Tried to call a non generic tool via the generic call method
A workflow error occurred, try to rollback / abort submitted jobs.
bkill 0x88FC0435A70E4 0x88FC045E1E94B 0x88FC0523BB7F4 0x88FC055016FF0 0x88FC057534853 0x88FC066054A97 8276 8277 8278 8279
An unknown / unhandled exception occurred: 'Not able to context tool coveragePlotSingle'
de.dkfz.roddy.knowledge.methods.GenericMethod._callGenericToolOrToolArray(GenericMethod.groovy:232)
de.dkfz.roddy.knowledge.methods.GenericMethod.callGenericTool(GenericMethod.groovy:50)
de.dkfz.b080.co.files.CoverageTextFile.plot(CoverageTextFile.java:49)
de.dkfz.b080.co.files.CoverageTextFileGroup.plot(CoverageTextFileGroup.java:47)
de.dkfz.b080.co.qcworkflow.QCPipeline.execute(QCPipeline.groovy:94)
de.dkfz.roddy.core.ExecutionContext.execute(ExecutionContext.groovy:625)
de.dkfz.roddy.core.Analysis.executeRun(Analysis.java:397)
de.dkfz.roddy.core.Analysis.executeRun(Analysis.java:341)
de.dkfz.roddy.core.Analysis.rerun(Analysis.java:229)
de.dkfz.roddy.client.cliclient.RoddyCLIClient.rerun(RoddyCLIClient.groovy:494)
de.dkfz.roddy.client.cliclient.RoddyCLIClient.parseStartupMode(RoddyCLIClient.groovy:114)
de.dkfz.roddy.Roddy.parseRoddyStartupModeAndRun(Roddy.java:688)
de.dkfz.roddy.Roddy.startup(Roddy.java:282)
de.dkfz.roddy.Roddy.main(Roddy.java:214)

There were errors for the execution context for dataset testpid
        * An uncaught error occurred during a run. SEVERE: An uncaught error occurred during a run.
```

bkill is apparently called on jobs that don't have a valid identifier.

**Comment by vinjana** [comment]
The message "Tried to call a non generic tool via the generic call method"  disappeared, after I added the previously forgotten 'overrideresourcesets="true"' to the resources XML. 

The subsequent errors disappeared, probably because the root cause for the rollback was removed. Whether bkill correctly works if incorrect IDs are provided should be tested, though!

**Comment by dankwart-de** [comment]
It should not be called, when testrun or rerun is called. We need to fix this.

**Comment by dankwart-de** [comment]
I guess it was solved with #157
